### ISSUE TITLE
Error: Guarantee failed: [token.has()]

### ISSUE THREAD

**Issue by bgnorlov** [rethinkdb/rethinkdb#5715]
I have some error on Ubuntu Xenial with rethinkdb 2.3.0 and 2.3.1 (compiled from source) when I tried to perform some aggregation query like ```r.table("users").orderBy("active_boards_count", {"index": r.desc("active_boards_count")}).count()``` 
```
error: Error in src/serializer/log/log_serializer.cc at line 438:
error: Guarantee failed: [token.has()]
error: Backtrace:
addr2line: 'rethinkdb': No such file
addr2line: 'rethinkdb': No such file
error: Sun Apr 24 05:50:57 2016

       1 [0xb2e69a]: backtrace_t::backtrace_t() at 0xb2e69a (rethinkdb)
       2 [0xb2eb7a]: format_backtrace[abi:cxx11](bool) at 0xb2eb7a (rethinkdb)
       3 [0xdf847c]: report_fatal_error(char const*, int, char const*, ...) at 0xdf847c (rethinkdb)
       4 [0xafee6c]: log_serializer_t::block_read(counted_t<ls_block_token_pointee_t> const&, file_account_t*) at 0xafee6c (rethinkdb)
       5 [0xaf71f5]: merger_serializer_t::block_read(counted_t<ls_block_token_pointee_t> const&, file_account_t*) at 0xaf71f5 (rethinkdb)
       6 [0xaf8935]: translator_serializer_t::block_read(counted_t<ls_block_token_pointee_t> const&, file_account_t*) at 0xaf8935 (rethinkdb)
       7 [0xaf5506]: alt::page_t::catch_up_with_deferred_load(alt::deferred_page_loader_t*, alt::page_cache_t*, cache_account_t*) at 0xaf5506 (rethinkdb)
       8 [0xa38812]: coro_t::run() at 0xa38812 (rethinkdb)
error: Exiting.
```

**Comment by danielmewes** [comment]
@bgnorlov Thank you for the bug report.
We'll look into this as soon as possible.

As a work-around, can you drop and then recreate the index `active_boards_count`?
I have a suspicion that this is an intermittent issue with index creation, so recreating the index might avoid the issue for now.

**Comment by bgnorlov** [comment]
I dropped and recreated index ```active_boards_count``` and still getting the same error. But when I dropped whole database_data directory and reconverted all my data the error gone away!

**Comment by danielmewes** [comment]
We found a solution to this. @nighelles is wrapping up the bug fix and this is going to ship in RethinkDB 2.3.2.
@bgnorlov Thanks again for reporting it.

**Comment by nighelles** [comment]
This is in next.
